ios/testmanagerd/testlistener_test.go

```diff
@@ -0,0 +1,170 @@
+		assert.Equal(t, "mymethod", testListener.testSuite.TestCases[0].MethodName)
+	})
+
+	t.Run("Check test case failure", func(t *testing.T) {
+		testListener := NewTestListener(io.Discard, io.Discard)
+
+		testListener.testSuiteDidStart("mysuite", "2024-01-16 15:36:43 +0000")
+		testListener.testCaseDidStartForClass("myclass", "mymethod")
+
+		testListener.testCaseFailedForClass("myclass", "mymethod", "error", "file://app.swift", 123)
+
+		assert.Equal(t, 1, len(testListener.testSuite.TestCases), "TestCase must be appended to list of test cases")
+		assert.Equal(t, "failed", testListener.testSuite.TestCases[0].Status.Status)
+		assert.Equal(t, "error", testListener.testSuite.TestCases[0].Status.Err.Message)
+		assert.Equal(t, "file://app.swift", testListener.testSuite.TestCases[0].Status.Err.File)
+		assert.Equal(t, uint64(123), testListener.testSuite.TestCases[0].Status.Err.Line)
+	})
+
+	t.Run("Check test case finish", func(t *testing.T) {
+		testListener := NewTestListener(io.Discard, io.Discard)
+
+		testListener.testSuiteDidStart("mysuite", "2024-01-16 15:36:43 +0000")
+		testListener.testCaseDidStartForClass("myclass", "mymethod")
+		testListener.testCaseDidFinishForTest("myclass", "mymethod", "passed", 1.0)
+
+		assert.Equal(t, 1, len(testListener.testSuite.TestCases), "TestCase must be appended to list of test cases")
+		assert.Equal(t, "passed", testListener.testSuite.TestCases[0].Status.Status)
+		assert.Equal(t, 1.0, testListener.testSuite.TestCases[0].Duration.Seconds())
+	})
+
+	t.Run("Check test suite finish", func(t *testing.T) {
+		testListener := NewTestListener(io.Discard, io.Discard)
+
+		testListener.testSuiteDidStart("mysuite", "2024-01-16 15:36:43 +0000")
+		testListener.testCaseDidStartForClass("myclass", "mymethod1")
+		testListener.testCaseDidFinishForTest("myclass", "mymethod1", "passed", 1.0)
+		testListener.testCaseDidStartForClass("myclass", "mymethod2")
+		testListener.testCaseDidFinishForTest("myclass", "mymethod2", "passed", 1.0)
+		testListener.testSuiteFinished("myclass", "2024-01-16 15:36:44 +0000", 2, 0, 0, 0, 0, 0, 1.0, 2.0)
+
+		assert.Equal(t, 2, len(testListener.testSuite.TestCases), "TestCase must be appended to list of test cases")
+		assert.Equal(t, "passed", testListener.testSuite.TestCases[0].Status.Status)
+		assert.Equal(t, "passed", testListener.testSuite.TestCases[1].Status.Status)
+		assert.Equal(t, 2.0, testListener.testSuite.TotalDuration.Seconds())
+		assert.Equal(t, 1, testListener.testSuite.EndDate.Second()-testListener.testSuite.StartDate.Second())
+	})
+
+	t.Run("Check test case stall", func(t *testing.T) {
+		testListener := NewTestListener(io.Discard, io.Discard)
+
+		testListener.testSuiteDidStart("mysuite", "2024-01-16 15:36:43 +0000")
+		testListener.testCaseDidStartForClass("myclass", "mymethod1")
+		testListener.testCaseStalled("myclass", "mymethod1", "file://app.swift", 123)
+		testListener.testCaseDidFinishForTest("myclass", "mymethod2", "failed", 1.0)
```

Comment:
That's already a the case I mentioned in the comment of `testCaseFailedForClass`. Doesn't that mean two test cases were executed?